### PR TITLE
fix(chunker): Don't underflow with wrong size_hint

### PR DESCRIPTION
**Pull request by aawsome**
see #377

### CODE REVIEW

File: crates/core/src/chunker.rs [hunk 1]
```diff
@@ -16,8 +16,8 @@ pub(super) mod constants {
     pub(super) const MIN_SIZE: usize = 512 * KB;
     /// The maximum size of a chunk.
     pub(super) const MAX_SIZE: usize = 8 * MB;
-    /// Buffer size used for reading.
-    pub(super) const BUF_SIZE: usize = 64 * KB;
+    /// Buffer size used for reading - TODO: Find out optimal size for best performance!
+    pub(super) const BUF_SIZE: usize = 4 * KB;
```
Comment (nardoor) [inline crates/core/src/chunker.rs:20]: common kernel PAGE_SIZE would also be my **guess** ^^

File: crates/core/src/chunker.rs [hunk 2]
```diff
@@ -165,7 +163,7 @@ impl<R: Read + Send> Iterator for ChunkIter<R> {
             self.pos += 1;
             self.rabin.slide(byte);
         }
-        self.size_hint -= vec.len();
+        self.size_hint = self.size_hint.saturating_sub(vec.len()); // size_hint can be too small!
```
Comment (nardoor) [inline crates/core/src/chunker.rs:166]: nice
do we want to also give an example of that happening here ? (like linux pseudo fs in /proc)

**Review by nardoor — APPROVED** [review]
:+1:

File: crates/core/src/chunker.rs (hunk 2 above)
Comment (aawsome) [reply crates/core/src/chunker.rs:166]: I think the main use case is just that between `stat`ing a file and opening it, there could be another process which has modified the file - potentially making it larger..
But I would not add an explicit example; it is simply preventing an underflow here which just makes sense...

File: crates/core/src/chunker.rs (hunk 1 above)
Comment (aawsome) [reply crates/core/src/chunker.rs:20]: Yes, but we don't have performance tests yet. Maybe there is a better one.
Also, we could find out the performance results differ on different systems/plattforms. Then we might want to expose this as a parameter to users..

**Pull request merged**